etc/cmake/options.cmake

```diff
@@ -176,6 +176,11 @@ if(OT_PLATFORM_NETIF)
     list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE=1")
 endif()
 
+option(OT_NCP_VENDOR_HOOK "Enable vendor hook for NCP")
```

Comment:
Is this still necessary?

Reply:
The keyword OPENTHREAD_ENABLE_NCP_VENDOR_HOOK is still necessary and defined here to have a additional abstraction layer. Other defines are done in the same way (e.g. OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE). It is used in /src/ncp/ncp_uart.cpp.

Reply:
from https://cmake.org/cmake/help/latest/command/option.html, it seems `option()` is just for defining `ON`/`OFF`, so I think we should move this to `src/ncp/CMakefile`. Thoughts?

Reply:
Originally we wanted to use the same variables that were used in https://github.com/openthread/openthread/blob/master/configure.ac#L688-L707, but were asked to combine it into one option/variable. We try to do the same as `--with-ncp-vendor-hook-source=<VENDOR_HOOK.CPP>` but via cmake and not via configure. Suggestions?

Reply:
I see the problem. I'm fine with this PR if it's working and I'm considering introducing a `ot-config` target to better support change OpenThread configuration in a deep subdirectory.

Reply:
I've changed the name of OT_NCP_VENDOR_HOOK to OT_NCP_VENDOR_HOOK_SOURCE to align with https://github.com/openthread/openthread/blob/master/configure.ac#L688-L707.

I've tested it locally with my openthread/zephyr device and it worked fine.